src/storage/mvcc/reader/reader.rs

```diff
@@ -269,9 +270,16 @@ impl<S: Snapshot> MvccReader<S> {
                     return Ok(TxnCommitRecord::OverlappedRollback { commit_ts });
                 }
                 return Ok(TxnCommitRecord::None {
-                    overlapped_write: Some(write),
+                    overlapped_write: Some(
+                        write
+                            .set_overlapped_rollback(true)
+                            .set_gc_fence(Some(gc_fence)),
+                    ),
```

Comment:
Why immediately set the overlapped rollback and gc fence here? The `get_txn_commit_record` is used to load the transaction's status from disk, it doesn't assume what the result will be used to do. Also the record might already has overlapped rollback or gc fence.

For example, this function is also used when committing a transaction but lock is not found, to check what's going on about the transaction. No rollback will happen here.

Reply:
I prefer the implementation one commit ago: store potential gc fence separately in `OverlappedWrite`.